src/Svg/Tag/Circle.php

```diff
@@ -17,10 +19,12 @@ class Circle extends Shape
     public function start($attributes)
     {
         if (isset($attributes['cx'])) {
-            $this->cx = $attributes['cx'];
+            $width = $this->document->getWidth();
+            $this->cx = Style::convertSize($attributes['cx'], $width);
         }
         if (isset($attributes['cy'])) {
-            $this->cy = $attributes['cy'];
+            $height = $this->document->getHeight();
+            $this->cy = Style::convertSize($attributes['cy'], $height);
         }
         if (isset($attributes['r'])) {
             $this->r = $attributes['r'];
```

Comment:
Now added. I added a `getDiagonal()` to the document since it looks like the diagonal is used in a few areas. 
For this I used the exact calculate [required in the spec](https://svgwg.org/svg2-draft/coords.html#Units).